docs/create-packages/native-files-in-net-packages.md

```diff
@@ -0,0 +1,88 @@
+
+If your assembly contains a lot of managed code, not just P/Invoke method definitions, you can save space by making the `ref/` assembly a [reference assembly](/dotnet/standard/assembly/reference-assemblies#generating-reference-assemblies), but be sure to include it under `ref/` in your package, not `lib/`.
+
+Therefore, there is no capability to provide different compile-time assemblies for different RIDs, you will need to provide the same API surface for all RIDs, and do runtime checks (for example, throw )
+
+### Runtime assets
+
+Runtime assets are copied to the output directory on build and publish, and the .NET SDK generates the `deps.json` with all the RID-specific asset information.
+NuGet will select runtime assets from `runtimes/{rid}/native/` and `runtimes/{rid}/lib/{tfm}/` (see below to understand `sub-path`), and if that doesn't exist, then it will fall back to `lib/{tfm}/`.
+
+When there are both `runtimes/{rid}/native/` and `runtimes/{rid}/lib/{tfm}/` files, NuGet will always select the `runtimes/{rid}/native/` assets, and then the assets from only one of the `runtimes/{rid}/lib/{tfm}/` directories, just like it does for compile assets in `ref/{tfms}/` or `lib/{tfms}/`.
+
+Similarly, NuGet selects the "best" RID, when multiple exist, or multiple match.
+Therefore, if a package contains both `runtimes/any/native/some.dll` and `runtimes/linux-x64/native/different.so`, when the project consuming the package targets `linux-x64`, NuGet will select `different.so`, and for all other RIDs that are not compatible with `linux-x64`, then NuGet will select `some.dll`.
+You cannot use the `any` RID to put common files that will be selected by all RIDs.
+
+## Putting it together
+
+How to build any binaries native libraries is out of scope of this document.
+It is assumed that all the native binaries and the .NET assemblies have been completed and compied to a machine for packing.
+
+Consider a package `Contoso.Native` that provides .NET APIs to `contoso.dll` on Windows, and `contoso.so` on other platforms.
+The package contents should be (excluding files required by the [Open Packaging Conventions](https://en.wikipedia.org/wiki/Open_Packaging_Conventions)):
+
+```text
+Contoso.Native.1.0.0.nuspec
+ref/net8.0/Contoso.Native.dll
+runtimes/android/lib/net8.0/Contoso.Native.dll
+runtimes/android/native/contoso.so
+runtimes/ios/lib/net8.0/Contoso.Native.dll
+runtimes/ios/native/contoso.so
+runtimes/linux-arm64/lib/net8.0/Contoso.Native.dll
+runtimes/linux-arm64/native/contoso.so
+runtimes/linux-x64/lib/net8.0/Contoso.Native.dll
+runtimes/linux-x64/native/contoso.so
+runtimes/maccatalyst-arm64/lib/net8.0/Contoso.Native.dll
+runtimes/maccatalyst-arm64/native/contoso.so
+runtimes/maccatalyst-x64/lib/net8.0/Contoso.Native.dll
+runtimes/maccatalyst-x64/native/contoso.so
```

Comment:
I believe macOS and iOS native libraries will most of the time have a `.dylib` file extension.